src/org/exist/xquery/FunctionCall.java

```diff
@@ -362,15 +362,20 @@ public void accept(ExpressionVisitor visitor) {
         visitor.visitFunctionCall(this);
     }
 
-    private class DeferredFunctionCallImpl extends DeferredFunctionCall {
+    private static class DeferredFunctionCallImpl extends DeferredFunctionCall {
```

Comment:
@dizzzz Inner classes should always be `static` unless they explicitly and directly and need access to the outer classes variable members.

However, that said, the approach @shabanovd is taking looks wrong here to me. He explicitly passed the outer class because he needs access to the members. I think it is better to either:

1) Not make the inner class static.
2) Make the inner class static, but do not access the member variables directly, instead call the accessor functions, then these could be mocked for testing. i.e. instead of `call.mySignature` use `call.getSignature()`